src/PostgREST/Plan.hs

```diff
@@ -1028,6 +1029,22 @@ callPlan proc ApiRequest{} paramKeys args readReq = FunctionCall {
             | otherwise            -> KeyParams $ specifiedParams [prm]
       prms  -> KeyParams $ specifiedParams prms
 
+-- | Get filter fields/column names from read plan
+getFilterFieldNames :: ReadPlanTree -> [FieldName]
+getFilterFieldNames = foldr (\rp names -> names ++ rpToFieldNames rp) []
+  where
+    rpToFieldNames :: ReadPlan -> [FieldName]
+    rpToFieldNames = logicTreesToFieldName . ReadPlan.where_
+
+    logicTreesToFieldName :: [CoercibleLogicTree] -> [FieldName]
+    logicTreesToFieldName = concatMap coLogicTreeToFieldNames
+
+    coLogicTreeToFieldNames :: CoercibleLogicTree -> [FieldName]
+    coLogicTreeToFieldNames = \case
+      CoercibleStmnt (CoercibleFilter{field=CoercibleField{cfName}}) -> [cfName]
+      CoercibleStmnt (CoercibleFilterNullEmbed _ cfName) -> [cfName]
+      CoercibleExpr _ _ clts -> concatMap coLogicTreeToFieldNames clts
```

Comment:
@laurenceisla I can't think of a test to `code-cov` these two lines? Any ideas?

Reply:
If I understand correctly, `CoercibleStmnt (CoercibleFilterNullEmbed ...)` is for `IS NULL` filters done to the embedded resource, like: `/rpc/getallprojects?select=client()&client=not.is.null`. The `CoercibleExpr` is for logical filters like: `/rpc/getallprojects?or=(id.eq.1,id.eq.2)` (just examples to illustrate, not sure if relevant to this PR).